ids/node_id.go

```diff
@@ -60,35 +52,60 @@ func (id *NodeID) UnmarshalJSON(b []byte) error {
 	}
 
 	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
+	*nodeID, err = NodeIDFromString(str[1:lastIndex])
 	return err
 }
 
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+func (nodeID NodeID) MarshalText() ([]byte, error) {
+	return []byte(nodeID.String()), nil
+}
+
+func (nodeID *NodeID) UnmarshalText(text []byte) error {
+	return nodeID.UnmarshalJSON(text)
 }
 
-func (id NodeID) Less(other NodeID) bool {
-	return bytes.Compare(id[:], other[:]) == -1
+func (nodeID NodeID) Less(other NodeID) bool {
+	return nodeID < other
 }
 
-// ToNodeID attempt to convert a byte slice into a node id
-func ToNodeID(bytes []byte) (NodeID, error) {
-	nodeID, err := ToShortID(bytes)
-	return NodeID(nodeID), err
+func (nodeID NodeID) Equal(other NodeID) bool {
+	return nodeID == other
+}
+
+func NodeIDFromBytes(bytes []byte) NodeID {
+	return NodeID(string(bytes))
```

Comment:
As per the earlier comment let's validate length here to avoid having to do the `ToSize` trick.